keps/sig-network/2433-topology-aware-hints/README.md

```diff
@@ -369,12 +376,27 @@ of the following scenarios:
 2. A new Node results in a Service that is able to achieve an endpoint
    distribution below 20% for the first time.
 
+### SameZone Heuristic
+The simplest of all heuristics, this will simply copy the zone of an endpoint to
+the hint of the same endpoint. If there are not endpoints available within a zone,
+that will mean no endpoints will have hints for that zone. As described above,
+when Kube-Proxy cannot find any endpoints with hints for its zone, it will route
+traffic ~evenly across all endpoints, regardless of hints.
+
+### Additional Heuristics
+To enable additional heuristics to be added in the future, we will:
+
+1. Remove the requirement in kube-proxy that the hints annotation must be set to
+   a known value on the associated Service before the values of EndpointSlice
+   hints will be considered.
+2. Ensure the EndpointSlice controller TopologyCache provides an interface that
```

Comment:
I was exploring this idea and I ended checking these options
- lua
- wasm
- grpc

Reply:
I was actually more thinking of at least one of the following

1) making it easier to run separate EPS controllers with custom logic (see https://github.com/kubernetes/enhancements/pull/3686)
2) providing a path to add experimental new heuristics to the core EPS controller

Reply:
yeah, I meant those are some technologies that can implement 2., or what were you thinking?

Reply:
I actually meant more feature-gated code that would be added in future Kubernetes releases. Essentially, if you want to do something really custom, run your own controller. If you want to build something that you think would be broadly useful, integrate it into k/k as a new EPS Controller heuristic.

Reply:
oh, ok, and how you enabled it? using a flag in the controller-manager?

Reply:
I was thinking like Rob, too.  Build it as a custom EPS controller, show some results, then PR it as a named heuristic to k/k and let people choose it by annotation.

Reply:
fair enough, whatever the implementation model we want, I really think this should be out of tree.